streaming/src/main/scala/org/apache/spark/streaming/dstream/FileInputDStream.scala

```diff
@@ -195,10 +195,13 @@ class FileInputDStream[K, V, F <: NewInputFormat[K, V]](
       )
       logDebug(s"Getting new files for time $currentTime, " +
         s"ignoring files older than $modTimeIgnoreThreshold")
-      val filter = new PathFilter {
+      val newFileFilter = new PathFilter {
         def accept(path: Path): Boolean = isNewFile(path, currentTime, modTimeIgnoreThreshold)
       }
-      val newFiles = fs.listStatus(directoryPath, filter).map(_.getPath.toString)
+      val directories = fs.globStatus(directoryPath, directoryFilter).map(_.getPath)
```

Comment:
Got it, right, so I can still specify a directory and it will match that directory only in the first globStatus. That makes sense. (Nit in the line below: you don't need the extra braces and will probably have to remove the space before 'dir'.)